src/components/form/HdInputPhone.vue

```diff
@@ -0,0 +1,383 @@
+
+const ALPHABET_UPPERCASE_CODES = [65, 90];
+const ALPHABET_LOWERCASE_CODES = [97, 122];
+
+const isAlphabetKey = (key) => {
+  const [A, Z] = ALPHABET_UPPERCASE_CODES;
+  const [a, z] = ALPHABET_LOWERCASE_CODES;
+  return (key >= A && key <= Z) || (key >= a && key <= z);
+};
+
+export default {
+  name: 'HdInputPhone',
+  mixins: [
+    formField,
+  ],
+  inheritAttrs: false,
+  components: {
+    HdInputFormatter,
+    HdIcon,
+  },
+  props: {
+    defaultCountry: {
+      type: String,
+      default: 'DE',
+    },
+    preferredCountries: {
+      type: Array,
+      default: () => ([]),
+    },
+    value: {
+      type: String,
+      default: '',
+    },
+    name: {
+      type: String,
+      required: true,
+    },
+    lang: {
+      type: String,
+      default: 'de',
+    },
+  },
+  data() {
+    return {
+      showDropdown: false,
+      phoneNumber: COUNTRY_PHONE_CODES.find((country) => country.code === this.defaultCountry)?.dial_code,
+      selectedCountry: COUNTRY_PHONE_CODES.find((country) => country.code === this.defaultCountry),
+      focusedCountry: null,
+      rule: {
+        validate: (value) => {
+          const phoneNumber = new PhoneNumber(value);
+          return phoneNumber.isValid() && phoneNumber.isMobile() && phoneNumber.canBeInternationallyDialled();
+        },
+        errorMessage: 'e.g. +55 555 555 5555',
+      },
+      smallArrowIcon,
+    };
+  },
+  computed: {
+    sortedCountriesList() {
+      // Preferred countries first
+      if (this.preferredCountries.length) {
+        return COUNTRY_PHONE_CODES.sort((a, b) => {
+          // Alphabetical order on preferred countries
+          if (this.preferredCountries.includes(a.code) && this.preferredCountries.includes(b.code)) {
+            return this.t.COUNTRIES[a.code] > this.t.COUNTRIES[b.code];
+          }
+          if (this.preferredCountries.includes(a.code)) {
+            return -1;
+          } if (this.preferredCountries.includes(b.code)) {
+            return 1;
+          // Alphabetical order on rest of countries
+          } if (!this.preferredCountries.includes(a.code) && !this.preferredCountries.includes(b.code)) {
+            return (this.t.COUNTRIES[a.code] > this.t.COUNTRIES[b.code]) ? 1 : -1;
+          }
+          return 0;
+        });
+      }
+      return COUNTRY_PHONE_CODES;
+    },
+    arrowClassNames() {
+      const baseClass = 'phone-input__selector__arrow';
+      const directionClass = this.showDropdown ? 'phone-input__selector__arrow--down' : 'phone-input__selector__arrow--up';
+      return [baseClass, directionClass];
+    },
+    selectedFlagClassNames() {
+      const baseClass = 'phone-input__selector__flag';
+      const countryClass = `flag-icon flag-icon-${this.selectedCountry.code.toLowerCase()}`;
+      return [baseClass, countryClass];
+    },
+    activeCountryCodeAndName() {
+      return `${this.selectedCountry.dial_code}, ${this.t.COUNTRIES[this.selectedCountry.code]}`;
+    },
```

Comment:
Oh I could have used a third party library for all the feature but at the beginning I didn't consider it because I felt that I was cheating 🙈 I didn't know that we were allowed to do that because it would make the library bigger (due to the dependencies), and also I thought that it was better if we could configure certain things on our own if those were too specific. I also wanted to use our own Input components as it already has a lot of logic for formatting and passing value, labels and other props.